### PR TITLE
check and install libffi-devel (point to redhat issue details)

### PR DESCRIPTION
**Pull request by dennybaa**

### CODE REVIEW

**Comment by dzimine** [comment]
+1

File: scripts/st2bootstrap-el6.sh [hunk 1]
```diff
@@ -3,18 +3,23 @@
...
-    echo "   rpm -qa | grep libffi  "
-    echo "Installer will now abort. Contact support for questions or see docs"
-    echo "   https://docs.stackstorm.com/install/rhel6.html!"
-    echo "Alternatively, you can use CentOS 6 for evaluation."
+  local message= no_libffi_devel=
+message=$(cat <<EHD
+No repository containing libffi-devel package has been located!
+Setup "server-optional" repository following instructions
+https://access.redhat.com/solutions/265523. After adding the repository using
+your preffered method (subscription or yum-utils) please re-run this script!
```
Comment (dennybaa) [inline scripts/st2bootstrap-el6.sh]: preferred

File: scripts/st2bootstrap-el6.sh [hunk 2]
```diff
@@ -3,18 +3,23 @@
...
+to use CentOS 6 platform if it's suitable for you.
+EHD
+)
+  yum list libffi-devel 1>/dev/null 2>&1 || no_libffi_devel=$?
+  if [ ! -z "$no_libffi_devel" ]; then
+    echo "$message"
     exit 2
+  else
+    sudo yum install -y libffi-devel
   fi
```
Comment (dennybaa) [inline scripts/st2bootstrap-el6.sh]: sudo yum install -y libffi-devel || exit 2

File: scripts/st2bootstrap-el6.sh (hunk 1 above)
Comment (lakshmi-kannan) [inline scripts/st2bootstrap-el6.sh]: yes, please fix.

File: scripts/st2bootstrap-el6.sh [hunk 3]
```diff
@@ -3,18 +3,23 @@
...
-    echo "   https://docs.stackstorm.com/install/rhel6.html!"
-    echo "Alternatively, you can use CentOS 6 for evaluation."
+  local message= no_libffi_devel=
+message=$(cat <<EHD
+No repository containing libffi-devel package has been located!
+Setup "server-optional" repository following instructions
+https://access.redhat.com/solutions/265523. After adding the repository using
+your preffered method (subscription or yum-utils) please re-run this script!
+
+If you still have questions please contact the support or also you can try
```
Comment (lakshmi-kannan) [inline scripts/st2bootstrap-el6.sh]: "If you still have questions, please contact support. Alternatively, you can use CentOS 6 for evaluation."

File: scripts/st2bootstrap-el6.sh (hunk 2 above)
Comment (lakshmi-kannan) [inline scripts/st2bootstrap-el6.sh]: We really don't need to install here as st2 installation will get that. I didn't know about yum list and that's why I tried to install and grep for "No package" in my PR. Did you validate yum list on RHEL 6.7 community AMI in AWS? yum search definitely didn't work there.

Comment (dennybaa) [inline scripts/st2bootstrap-el6.sh]: well you are right! no need install, just copy pasted yours not thinking, lol)

Comment (dennybaa) [inline scripts/st2bootstrap-el6.sh]: `yum list` is a standard command it looks through local db. available since ages on any rhel and centos

Comment (dennybaa) [inline scripts/st2bootstrap-el6.sh]: @lakshmi-kannan https://github.com/StackStorm/st2docs/commit/f1d4da9255a786f6ef8bf31c35aeadfd65f73c59#commitcomment-16383737

**Pull request merged**